modules/microservice/main.tf

```diff
@@ -757,7 +758,7 @@ locals {
   function_app_endpoint_resources = local.http_target == "function" ? { for function in azurerm_function_app.microservice : function.location => { id = function.id, location = function.location } } : {}
   azure_endpoint_resources        = merge(local.app_service_endpoint_resources, local.function_app_endpoint_resources)
 
-  static_endpoint_primary_resources   = local.has_static_site ? { for site in azurerm_storage_account.microservice : site.primary_web_host => { id = site.id, host = site.primary_web_host } if site.static_website != null } : {}
-  static_endpoint_secondary_resources = local.has_static_site ? { for site in azurerm_storage_account.microservice : site.secondary_web_host => { id = site.id, host = site.secondary_web_host } if site.static_website != null } : {}
+  static_endpoint_primary_resources   = { for site in azurerm_storage_account.microservice : "${site.name}-primary" => { id = "${site.name}-primary", host = "${site.name}${local.staticsite_baseurl}" }} 
```

Comment:
Have you tried using this with name as the key and the dynamic url? I believe you can structure this slightly differently and still fix the issue without needing to hard code the url. I started thinking more about this after seeing the ".z5." portion hard coded

Reply:
I did, unfortunately.  Most of the storage account attributes are not available until after the apply which is the root of this issue.

Reply:
Glad we were able to work through it together! :)